src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -1758,12 +1762,44 @@ SendQueryInSingleRowMode(MultiConnection *connection, char *query,
 
 
 /*
- * ExtractParametersFromParamListInfo extracts parameter types and values from
+ * ExtractParametersForRemoteExecution extracts parameter types and values from
  * the given ParamListInfo structure, and fills parameter type and value arrays.
+ * It changes oid of custom types to InvalidOid so that they are the same in workers
+ * and coordinators.
  */
 void
-ExtractParametersFromParamListInfo(ParamListInfo paramListInfo, Oid **parameterTypes,
+ExtractParametersForRemoteExecution(ParamListInfo paramListInfo, Oid **parameterTypes,
+									const char ***parameterValues)
+{
+	ExtractParametersInternal(paramListInfo, parameterTypes,
+							  parameterValues, false);
+}
+
+
+/*
+ * ExtractParametersForLocalExecution extracts parameter types and values from
+ * the given ParamListInfo structure, and fills parameter type and value arrays.
+ * It does not change the oid of custom types, because the query will be run locally.
+ */
+void
+ExtractParametersForLocalExecution(ParamListInfo paramListInfo, Oid **parameterTypes,
 								   const char ***parameterValues)
+{
+	ExtractParametersInternal(paramListInfo, parameterTypes,
+							  parameterValues, true);
+}
+
+
+/*
+ * ExtractParametersInternal extracts parameter types and values from
+ * the given ParamListInfo structure, and fills parameter type and value arrays.
+ * If isLocalExecution is true, it does not change the oid of custom types.
+ */
+void
+ExtractParametersInternal(ParamListInfo paramListInfo,
+						  Oid **parameterTypes,
+						  const char ***parameterValues, bool
+						  isLocalExecution)
```

Comment:
The name of this variable describes something about the caller, rather than about what its effect is. Maybe something along the lines of `setCustomTypeOids`? Possible uses can be described in the comments.

Reply:
I have changed it to `useOriginalCustomTypeOids`